lib/ash/resource/transformers/require_alphanumeric_field_names.ex

```diff
@@ -0,0 +1,49 @@
+defmodule Ash.Resource.Transformers.RequireAlphanumericFieldNames do
+  @moduledoc """
+  Confirms that a resource does not have unaccepted characters (characters other than alphanumeric and underscores) in field names.
+  """
+  use Spark.Dsl.Transformer
+
+  alias Spark.Dsl.Transformer
+  alias Spark.Error.DslError
+
+  def transform(dsl_state) do
+    attributes =
+      dsl_state
+      |> Transformer.get_entities([:attributes])
+
+    relationships =
+      dsl_state
+      |> Transformer.get_entities([:relationships])
+
+    calculations =
+      dsl_state
+      |> Transformer.get_entities([:calculations])
+
+    aggregates =
+      dsl_state
+      |> Transformer.get_entities([:aggregates])
+
+    attributes
+    |> Enum.concat(relationships)
+    |> Enum.concat(calculations)
+    |> Enum.concat(aggregates)
+    |> Enum.group_by(& &1.name)
+    |> Enum.each(fn {name, _groups} ->
+      unless name_acceptable?(name) do
+        raise DslError.exception(message: "The field name #{name} contains special characters")
```

Comment:
The message is a bit placeholdery and not too informative (it doesn't say what "special characters" are). What would you hope to see in it?

Reply:
A couple things here: 

So calculations/aggregates/relationships/attributes all have to have unique names anyway, so I wouldn't worry about that.

What you can do is something like this:
```elixir
   resource = Transformer.get_persisted(dsl_state, :module)

    attributes
    |> Enum.concat(relationships)
    |> Enum.concat(calculations)
    |> Enum.concat(aggregates)
    |> Enum.each(fn  %{name: name} = field -> 
      path = 
        case field do
          %Ash.Resource.Aggregate{type: type} -> [:aggregates, type, name]
          %Ash.Resource.Calculation{} -> [:calculations, name]
          %Ash.Resource.Attribute{} -> [:attributes, name]
          %{type: type} -> [:relationships, type, name] 
        end
      
      raise DslError, message: "your message is good, but add the regex and/or just say the rules", path: path, module: resource
    end)
```

This sets a path & module to help point people where the error is.

Additionally, instead of `def after?(...)` in your case this is just a validation that can be run in the after compile step, so remove that and add `def after_compile?, do: true`, and then you're good 🔥